google/api_core/client_logging.py

```diff
@@ -0,0 +1,89 @@
+import logging
+import json
+import re
+import os
+
+_LOGGING_INITIALIZED = False
+_BASE_LOGGER_NAME = "google"
+# TODO(https://github.com/googleapis/python-api-core/issues/760): Update Request / Response messages.
+REQUEST_MESSAGE = "Sending request "
+RESPONSE_MESSAGE = "Receiving response "
```

Comment:
Consider keeping these in `gapic-generator-python` to avoid having to require newer versions of `google-api-core` when log events are added downstream

Reply:
You mean during development, if we modifiy the generator before we release `google-api-core`? We will need to require updated versions of the `google-api-core` when we release the logging-enabled GAPICs. Otherwise, log messages from the GAPICs will propagate to the root logger, which we don't want.

Reply:
> We will need to require updated versions of the google-api-core when we release the logging-enabled GAPICs.

My initial thought is that we don't want to force users to update to the latest version of `google-api-core`. We can detect whether a newer version of `google-api-core` is installed, and avoid logging when there is an old version.


We don't want to force users onto newer versions of google-api-core every time that we want to define a new log. Ideally users can remain on old versions of `google-api-core` and only update as new functionality is needed (for example, if they choose to enable logging, they can only do so with a newer version of `google-api-core`).

Reply:
I think the setup defined here and any new log events that we add in our gapic should be guarded.

Reply:
> We don't want to force users onto newer versions of google-api-core every time that we want to define a new log. Ideally users can remain on old versions of google-api-core and only update as new functionality is needed (for example, if they choose to enable logging, they can only do so with a newer version of google-api-core).

I don't follow. I'm not suggesting users update `python-api-core` "every time that we want to define a new log". I'm suggesting they update ONCE, with essentially the functionality in this PR. New GAPICs will depend on this functionality regardless of where/when we add the log statements.

> any new log events that we add in our gapic should be guarded.

If we have the GAPICs depend on this version of `core`, then they won't be needed to be guarded, which makes the generator changes more sustainable. (Regardless, any guarding we do should be well encapsulated so logging an event remains a one liner.)

Reply:
What happens when we want to add another message like `REQUEST_MESSAGE = "Sending request "`, for example, `REQUEST_MESSAGE_METADATA = "Initial Metadata: "`? Users will need to upgrade `google-api-core` to be able to see those logs.